src/app/report/overwrite-transformation-modal/overwrite-transformation-modal.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, inject, output, TemplateRef, ViewChild } from '@angular/core';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-overwrite-transformation-modal',
+  standalone: true,
+  templateUrl: './overwrite-transformation-modal.component.html',
+  styleUrl: './overwrite-transformation-modal.component.css',
+})
+export class OverwriteTransformationComponent {
+  @ViewChild('modal') protected modal!: TemplateRef<OverwriteTransformationComponent>;
+  confirm = output<void>();
+  protected activeModal?: NgbModalRef;
+  private modalService = inject(NgbModal);
+
+  open(): void {
+    this.activeModal = this.modalService.open(this.modal, {
```

Comment:
Normaal hoor je dit in de component te doen waar het modal doet openen, zo hoef je in de html ook niet alles in een ng-template te wrappen